diff --git a/DisplayHardware/HWComposer.cpp b/DisplayHardware/HWComposer.cpp
old mode 100644
new mode 100755
index 854f572..e706107
--- a/DisplayHardware/HWComposer.cpp
+++ b/DisplayHardware/HWComposer.cpp
@@ -267,7 +267,7 @@ void HWComposer::dump(String8& result, char* buffer, size_t SIZE,
             }
             result.appendFormat(
                     " %8s | %08x | %08x | %08x | %02x | %05x | %08x | [%5d,%5d,%5d,%5d] | [%5d,%5d,%5d,%5d] %s\n",
-                    l.compositionType ? "OVERLAY" : "FB",
+                    l.compositionType >= 1 ? (l.compositionType == 2 ? "CopyBit" : "OverLay") : "OpenGL",
                     intptr_t(l.handle), l.hints, l.flags, l.transform, l.blending, format,
                     l.sourceCrop.left, l.sourceCrop.top, l.sourceCrop.right, l.sourceCrop.bottom,
                     l.displayFrame.left, l.displayFrame.top, l.displayFrame.right, l.displayFrame.bottom,
